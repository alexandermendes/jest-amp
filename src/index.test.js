import { toBeValidAmpHtml } from '.';

describe('jest-amp', () => {
  describe('toBeValidAmpHtml', () => {
    const body = '<div>Hello, World!</div>';

    const successfulResult = { errors: [] };

    const failingResult = {
      errors: [
        {
          message: 'No good',
          specUrl: 'http://example.com/no-good',
        },
        {
          message: 'Also no good',
          specUrl: 'http://example.com/also-no-good',
        },
      ],
    };

    it('passes when there are no validation errors', () => {
      const { pass } = toBeValidAmpHtml({ result: successfulResult, body });

      expect(pass).toEqual(true);
    });

    it('does not pass when there are validation errors', () => {
      const { pass, message } = toBeValidAmpHtml({ result: failingResult, body });

      expect(pass).toEqual(false);
      expect(message()).toMatchSnapshot();
    });

    it('reports all errors', () => {
      const { message } = toBeValidAmpHtml({ result: failingResult, body });
      const output = message();

      expect(output).toMatch(/.*No good.*/);
      expect(output).toMatch(/.*Also no good.*/);
    });
  });
});
