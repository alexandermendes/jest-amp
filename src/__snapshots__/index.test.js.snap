// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-amp toBeValidAmpHtml does not pass when there are validation errors 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toBeValidAmpHtml([22m[32mexpected[39m[2m)[22m

Expected valid AMP HTML:

[31m\\"No good\\"[39m
http://example.com/no-good

[31m\\"Also no good\\"[39m
http://example.com/also-no-good

----------

<div>
  Hello, World!
</div>

----------"
`;
